### PR TITLE
API Enabled members report is now performant, no count column/filter, extra method filter options

### LINKED ISSUE #314: Member MFA report: remove number of registered methods filter
We've recently implemented a member report to review details of MFA. There's a filter option to filter the results by 'no. of registered methods'.

This seems like an unlikely option to filter by, and it's expected to cause performance issues at scale. As a result, this issue is created to remove that filter.

ACs
- Filter is removed
- Report is tested at scale to confirm performance is still sufficient

### PR DESCRIPTION
**Pull request by robbieaverill**
Fixes #314

### CODE REVIEW

File: src/Report/EnabledMembers.php [hunk 1]
```diff
@@ -114,12 +92,22 @@ public function sourceRecords($params): SS_List
...
+                'title' => _t(__CLASS__ . '.COLUMN_METHODS_REGISTERED', 'Registered methods'),
+                'formatting' => [$this, 'formatMethodsColumn']
+            ],
+            'defaultMethodName' => [
+                'title' => _t(__CLASS__ . '.COLUMN_METHOD_DEFAULT', 'Default method'),
+                'formatting' => [$this, 'formatDefaultMethodColumn']
+            ],
+            'HasSkippedMFARegistration' => [
+                'title' => _t(__CLASS__ . '.COLUMN_SKIPPED_REGISTRATION', 'Skipped registration'),
+                'formatting' => function ($_, $record) { return $record->HasSkippedMFARegistration ? 'Yes' : 'No'; }
```
Comment (ScopeyNZ) [inline src/Report/EnabledMembers.php]: Should've known this would make the linter unhappy...

File: src/Report/EnabledMembers.php [hunk 2]
```diff
@@ -130,84 +118,134 @@ public function columns(): array
...
+     * @param null $_
+     * @param Member $record
+     * @return string
+     */
+    public function formatMethodsColumn($_, Member $record): string
+    {
+        /** @var Member&MemberExtension $record */
+        $methods = $this->getRegisteredMethodsForRecords();
+
+        return implode(', ', array_map(function(RegisteredMethod $method) {
```
Comment (ScopeyNZ) [inline src/Report/EnabledMembers.php]:
```suggestion
        return implode(', ', array_map(function (RegisteredMethod $method) {
```

File: src/Report/EnabledMembers.php (hunk 1 above)
Comment (ScopeyNZ) [reply src/Report/EnabledMembers.php]:
```suggestion
                'formatting' => function ($_, $record) { 
                    return $record->HasSkippedMFARegistration ? 'Yes' : 'No'; 
                },
```

**Comment by robbieaverill** [comment]
Oh, let me write a test or two for the extra filters on method names

**Pull request merged**